kimchi/src/circuits/polynomials/range_check/witness.rs

```diff
@@ -151,35 +152,45 @@ const fn range_check_row(row: usize) -> [WitnessCell; COLUMNS] {
     ]
 }
 
-fn value_to_limb<F: PrimeField>(fe: F, start: usize, end: usize) -> F {
+pub fn value_to_limb<F: PrimeField>(fe: F, start: usize, end: usize) -> F {
     F::from_bits(&fe.to_bits()[start..end]).expect("failed to deserialize field bits")
 }
 
+pub fn handle_standard_witness_cell<F: PrimeField>(
+    witness: &mut [Vec<F>; COLUMNS],
+    witness_cell: &WitnessCell,
+    row: usize,
+    col: usize,
+    value: F,
+) {
+    match witness_cell {
+        WitnessCell::Copy(copy_cell) => {
+            witness[col][row] = witness[copy_cell.col][copy_cell.row];
+        }
+        WitnessCell::Value => {
+            witness[col][row] = value;
+        }
+        WitnessCell::Limb(limb_cell) => {
+            witness[col][row] = value_to_limb(
+                witness[limb_cell.col][limb_cell.row], // limb cell (row, col)
+                limb_cell.start,                       // starting bit
+                limb_cell.end,                         // ending bit (exclusive)
+            );
+        }
+        WitnessCell::Zero => {
+            witness[col][row] = F::zero();
+        }
+    }
+}
+
 fn init_range_check_row<F: PrimeField>(witness: &mut [Vec<F>; COLUMNS], row: usize, value: F) {
     for col in 0..COLUMNS {
-        match &WITNESS_SHAPE[row][col] {
-            WitnessCell::Copy(copy_cell) => {
-                witness[col][row] = witness[copy_cell.col][copy_cell.row];
-            }
-            WitnessCell::Value => {
-                witness[col][row] = value;
-            }
-            WitnessCell::Limb(limb_cell) => {
-                witness[col][row] = value_to_limb(
-                    witness[limb_cell.col][limb_cell.row], // limb cell (row, col)
-                    limb_cell.start,                       // starting bit
-                    limb_cell.end,                         // ending bit (exclusive)
-                );
-            }
-            WitnessCell::Zero => {
-                witness[col][row] = F::zero();
-            }
-        }
+        handle_standard_witness_cell(witness, &WITNESS_SHAPE[row][col], row, col, value);
```

Comment:
why this change?

Reply:
Because in FFAdd and FFMul we will have both FF-related cells and range-check witness cells. Then for the latter type, we wanted to call that piece of code that handles the range-check cells independently. But I agree that for the scope of this PR, this could be postponed to FFAdd. I will revert it.